src/main/java/it/pagopa/pn/paperchannel/service/impl/QueueListenerServiceImpl.java

```diff
@@ -82,11 +82,39 @@ public void internalListener(PrepareAsyncRequest body, int attempt) {
                         .doOnError(throwable -> {
                             log.error(throwable.getMessage());
                             if (throwable instanceof PnAddressFlowException) return;
+                            if (throwable instanceof PnF24FlowException pnF24FlowException) f24ErrorListener(pnF24FlowException.getF24Error(), pnF24FlowException.getF24Error().getAttempt());
+
                             throw new PnGenericException(PREPARE_ASYNC_LISTENER_EXCEPTION, PREPARE_ASYNC_LISTENER_EXCEPTION.getMessage());
                         }))
                 .block();
     }
 
+
+    public void f24ErrorListener(F24Error entity, Integer attempt) {
+        log.info("Start async for {} request id", entity.getRequestId());
+
+        String PROCESS_NAME = "F24 retry listener logic";
+        MDC.put(MDCUtils.MDC_PN_CTX_REQUEST_ID, entity.getRequestId());
+        log.logStartingProcess(PROCESS_NAME);
+        MDCUtils.addMDCToContextAndExecute(requestDeliveryDAO.getByRequestId(entity.getRequestId(), true)
+                        .flatMap(f24Service::preparePDF)
+                        .doOnSuccess(pnRequestError -> log.logEndingProcess(PROCESS_NAME))
+                        .doOnError(throwable ->  log.logEndingProcess(PROCESS_NAME, false, throwable.getMessage()))
+                        .onErrorResume(ex -> {

```

Comment:
esternalizzare in un metodo il body dell'onErrorResume in modo tale da utilizzarlo in internalListener